native/src/main/java/io/ballerina/stdlib/sql/nativeimpl/CallProcessor.java

```diff
@@ -229,133 +231,144 @@ private static void setCallParameters(Connection connection, CallableStatement s
-                    break;
-                case Types.SMALLINT:
-                    result = resultParameterProcessor.processSmallInt(statement, paramIndex);
-                    break;
-                case Types.INTEGER:
-                    result = resultParameterProcessor.processInteger(statement, paramIndex);
-                    break;
-                case Types.BIGINT:
-                    result = resultParameterProcessor.processBigInt(statement, paramIndex);
-                    break;
-                case Types.REAL:
-                    result = resultParameterProcessor.processReal(statement, paramIndex);
-                    break;
-                case Types.FLOAT:
-                    result = resultParameterProcessor.processFloat(statement, paramIndex);
-                    break;
-                case Types.DOUBLE:
-                    result = resultParameterProcessor.processDouble(statement, paramIndex);
-                    break;
-                case Types.NUMERIC:
-                    result = resultParameterProcessor.processNumeric(statement, paramIndex);
-                    break;
-                case Types.DECIMAL:
-                    result = resultParameterProcessor.processDecimal(statement, paramIndex);
-                    break;
-                case Types.BIT:
-                    result = resultParameterProcessor.processBit(statement, paramIndex);
-                    break;
-                case Types.BOOLEAN:
-                    result = resultParameterProcessor.processBoolean(statement, paramIndex);
-                    break;
-                case Types.REF:
-                case Types.REF_CURSOR:
-                    result = resultParameterProcessor.processRef(statement, paramIndex);
-                    // This is to clean up the result set attached to the ref cursor out parameter
-                    // when procedure call result is closed.
-                    procedureCallResult.addNativeData(Constants.REF_CURSOR_VALUE_NATIVE_DATA, result);
-                    break;
-                case Types.STRUCT:
-                    result = resultParameterProcessor.processStruct(statement, paramIndex);
-                    break;
-                case Types.SQLXML:
-                    result = resultParameterProcessor.processXML(statement, paramIndex);
-                    break;
-                default:
-                    result = resultParameterProcessor.processCustomOutParameters(statement, paramIndex, sqlType);
-            }
-            parameter.addNativeData(Constants.ParameterObject.VALUE_NATIVE_DATA, result);
+            parameter.addNativeData(RESULT_PARAMETER_PROCESSOR, resultParameterProcessor);
+            parameter.addNativeData(STATEMENT_NATIVE_DATA_FIELD, statement);
+            parameter.addNativeData(PROCEDURE_CALL_RESULT, procedureCallResult);
+        }
+    }
+
+    public static void populateOutParameter(BObject parameter) throws SQLException, ApplicationError {
```

Comment:
Shall we move this function to `OutParameterProcessor` and make it private?

Reply:
Fixed with: [1b6c29c](https://github.com/ballerina-platform/module-ballerina-sql/pull/739/commits/1b6c29c801e042c27d11104fc8dfe0d50ce9242f)